exec/vm.go

```diff
@@ -70,17 +71,30 @@ type VM struct {
 	RecoverPanic bool
 
 	abort bool // Flag for host functions to terminate execution
+
+	nativeBackend *nativeCompiler
 }
 
 // As per the WebAssembly spec: https://github.com/WebAssembly/design/blob/27ac254c854994103c24834a994be16f74f54186/Semantics.md#linear-memory
 const wasmPageSize = 65536 // (64 KB)
 
 var endianess = binary.LittleEndian
 
-// NewVM creates a new VM from a given module. If the module defines a
-// start function, it will be executed.
-func NewVM(module *wasm.Module) (*VM, error) {
+// VMOptions describes configuration for a VM.
+type VMOptions struct {
+	// EnableAOT enables ahead-of-time compilation of supported opcodes
+	// into runs of native instructions, if wagon supports native compilation
+	// for the current architecture.
+	EnableAOT bool
+}
+
+// NewVMWithOptions creates a new VM from a given module and options. If the module defines
+// a start function, it will be executed.
+func NewVMWithOptions(module *wasm.Module, options *VMOptions) (*VM, error) {
```

Comment:
I'd rather use the "functional options" pattern than this exported `VMOptions` struct.

_e.g._
```go
type VMOption func(c *config)

func EnableAOT(v bool) VMOption {
    return func(c *config) {
        c.EnableAOT = v
    }
}

func NewVM(module *wasm.Module, opts ...VMOption) (*VM, error) {
...
}
```

what do you think?

Reply:
Good idea!